server/rest/src/main/java/org/infinispan/rest/ALPNHandler.java

```diff
@@ -76,22 +78,56 @@ public void configurePipeline(ChannelPipeline pipeline, String protocol) {
     */
    protected void configureHttp2(ChannelPipeline pipeline) {
       pipeline.addLast(getHttp11To2ConnectionHandler());
+      addHttp1Handlers(pipeline);
+   }
+
+   /**
+    * Adds the commons Http1 handlers to the pipeline.
+    */
+   private void addHttp1Handlers(ChannelPipeline pipeline) {
+      // Handles http content encoding (gzip)
+      pipeline.addLast(new HttpContentCompressor(restServer.getConfiguration().getCompressionLevel()));
+      // Handles chunked data
+      pipeline.addLast(new HttpObjectAggregator(maxContentLength()));
+      // Handles Http/2 headers propagation from request to response
```

Comment:
well, it says HTTP2... does it work with HTTP1?

Reply:
Our handlers act as if they were dealing in HTTP/1 only, Netty provides a series of adaptors from HTTP2 to HTTP1 and back when the connection is HTTP/2. The only thing we should do is to send this streamId if it is present. It won't be in HTTP1 connections so this new handler becomes a no-op